src/graph/GoExecutor.cpp

```diff
@@ -668,162 +737,195 @@ void GoExecutor::onEmptyInputs() {
+        auto *eschema = resp.get_edge_schema();
+        if (eschema != nullptr) {
+            std::transform(eschema->cbegin(), eschema->cend(),
+                           std::inserter(edgeSchema, edgeSchema.begin()), [](auto &schema) {
+                               return std::make_pair(
+                                   schema.first,
+                                   std::make_shared<ResultSchemaProvider>(schema.second));
+                           });
         }
 
-        for (auto &vdata : resp.vertices) {
-            std::unique_ptr<RowReader> vreader;
-            // TODO(simon.liu) In issue #192, I will solve this problem for better.
-            if (!resp.__isset.vertices || resp.get_vertices() == nullptr) {
-                continue;
-            }
+        if (tagSchema.empty() && edgeSchema.empty()) {
+            continue;
+        }
 
-            if (vschema != nullptr) {
-                DCHECK(vdata.__isset.vertex_data);
-                vreader = RowReader::getRowReader(vdata.vertex_data, vschema);
-            }
+        for (auto &vdata : resp.vertices) {
             DCHECK(vdata.__isset.edge_data);
-            DCHECK(eschema != nullptr);
-            RowSetReader rsReader(eschema, vdata.edge_data);
-            auto iter = rsReader.begin();
-            while (iter) {
-                auto &getters = expCtx_->getters();
-                getters.getAliasProp = [&](const std::string &,
-                                           const std::string &prop) -> OptVariantType {
-                    auto res = RowReader::getPropByName(&*iter, prop);
-                    if (ok(res)) {
-                        return value(res);
-                    }
-                    return Status::Error("get edge prop failed");
-                };
-                getters.getSrcTagProp = [&](const std::string &tagName,
-                                            const std::string &prop) -> OptVariantType {
-                    auto tagIter = this->srcTagProps_.find(std::make_pair(tagName, prop));
-                    if (tagIter == this->srcTagProps_.end()) {
-                        auto msg = folly::sformat(
-                            "Src tagName : {} , propName : {} is not exist", tagName, prop);
-                        LOG(ERROR) << msg;
-                        return Status::Error(msg);
-                    }
-                    auto index = tagIter->second;
-                    const nebula::cpp2::ValueType &type = vschema->getFieldType(index);
-                    if (type == CommonConstants::kInvalidValueType()) {
-                        auto msg =
-                            folly::sformat("Tag: {} no schema for the index {}", tagName, index);
-                        LOG(ERROR) << msg;
-                        return Status::Error(msg);
-                    }
-                    auto res = RowReader::getPropByIndex(vreader.get(), index);
```

Comment:
I notice you change the logic from getPropByIndex to getPropByName,   the performance will be hurt.  Are you OK with it @dutor